src/main/java/net/dv8tion/jda/bot/sharding/ShardManager.java

```diff
@@ -100,6 +101,53 @@ default void addEventListeners(final IntFunction<Object> eventListenerProvider)
         });
     }
 
+    /**
+     * Remove all listeners from shards for which id's the provided filter returns {@code true}.
+     * The filter tests shard ids, and for those ids that it returns {@code true} for, the corresponding shards will
+     * have all their listeners removed.
+     *
+     * @param shardFilter
+     *        a filter that returns {@code true} for those shard ids that should have all their listeners removed
+     *
+     * @throws java.lang.IllegalArgumentException
+     *         If the provided shard filter is {@code null}.
+     */
+    default void removeEventListeners(final IntPredicate shardFilter)
```

Comment:
I don't like this method, I suggested something different on discord `removeEventListeners(IntFunction<Collection<Object>>)`